containers/base/javascript/browser/js/main.js

```diff
@@ -86,9 +108,10 @@ define([
 
     function activateElement(templatesContainer, klass, targetContainer, markerComment) {
         console.log('activateElement', templatesContainer, klass, targetContainer, markerComment);
-        var template = templatesContainer.getElementsByClassName(klass)[0]
-          , activatedElement = template.cloneNode(true)
-          ;
+        var templateString = templatesContainer.innerHTML;
+        var template = document.createRange().createContextualFragment(templateString);
+        var activatedElement = template.querySelector("." + klass);
```

Comment:
In this way some things like a simple querySelector work better. In the previous approach two instances of the same element existed. (Because of the cloning) This complicated a lot of things. It also paves the way for Material Design, which is initialised on page load. (In the old approach it connects to the version that is not yet visible.)

Reply:
Ok, so this is pretty cool. I thought `templatesContainer.innerHTML` would serialize the contents of `templatesContainer`, but since it's a `<script type="text/templates"` tag, the contents are not parsed in the first place and `innerHTML` returns the equivalent of `textContent`. Totally agree that it is better this way. I'll look into creating the `template` node only once and reuse it, (then use clone again) but it's fine so far, not having the template in the DOM is really good. Also to be able to use links in the document with ids, this is a must have.